### PR TITLE
Fix auto-completion code

### LINKED ISSUE #2816: unsolicited console interaction "Install mc auto-completion in your shell ?"
(I rename execitable to mcli because I use midnight-commander a lot)
REPRO:
curl -sSLo /usr/bin/mcli https://dl.minio.io/client/mc/release/linux-amd64/mc
chmod +x /usr/bin/mcli
EXPECTED:
can use mcli in scripts
CURENT:
it always goes interactive
mcli
mcli: Configuration written to `/root/.mc/config.json`. Please update your access credentials.
mcli: Successfully created `/root/.mc/share`.
mcli: Initialized share uploads `/root/.mc/share/uploads.json` file.
mcli: Initialized share downloads `/root/.mc/share/downloads.json` file.
Install mc auto-completion in your shell ? (y/n):

### PR DESCRIPTION
**Pull request by harshavardhana**
Reverts the code in #2812

Fixes #2816

### CODE REVIEW

**Comment by sinhaashish** [comment]
We should add the details for `autocompletion` below the --insecure section in
https://github.com/minio/mc/blob/master/docs/minio-admin-complete-guide.md#option----insecure
https://github.com/minio/mc/blob/master/docs/minio-client-complete-guide.md#option----insecure

**Comment by harshavardhana** [comment]
It is not a global option to be used always @sinhaashish, that's why it's left out of those docs. This command gets its own mention in our README.md and also as part of a new TIP in our command document.

File: cmd/main.go [hunk 1]
```diff
@@ -221,44 +222,22 @@ func initMC() {
...
-	}
-
+func installAutoCompletion() {
 	if runtime.GOOS == "windows" {
+		console.Infoln("autocompletion feature is not available for this operating system")
 		return
 	}
 
-	if completeinstall.IsInstalled("mc") {
+	if completeinstall.IsInstalled(filepath.Base(os.Args[0])) || completeinstall.IsInstalled("mc") {
```
Comment (vadmeste) [inline cmd/main.go]: The second completeinstall.IsInstalled("mc") doesn't seem to be useful here.. completion won't work anyway if the binary name is different to 'mc'

File: cmd/main.go [hunk 2]
```diff
@@ -221,44 +222,22 @@ func initMC() {
...
+		console.Infoln("autocompletion is already enabled in your '$SHELLRC'")
```
Comment (vadmeste) [inline cmd/main.go]: Why not simply 'shell' ? I mean did you intend to have a variable substitution here '$SHELLRC'  ?

File: cmd/main.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/main.go]: It will work it's already fixed.

File: cmd/main.go (hunk 2 above)
Comment (harshavardhana) [reply cmd/main.go]: That was AB's recommendation @vadmeste

**Comment by deekoder** [comment]
swapping @nitisht to @poornas  to complete approval and make release today.

**Pull request merged**